components/Product/basic_usage.rst

```diff
@@ -0,0 +1,163 @@
+Basic Usage
+===========
+
+Creating a product
+------------------
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Product\Model\Product;
+
+   $product = new Product();
+
+   $product->getCreatedAt(); // returns the \DateTime when it was created
+   $product->getAvailableOn(); // by default returns the same value as getCreatedAt
+
+Setting the archetype
+---------------------
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Product\Model\Archetype;
+   use Sylius\Component\Product\Model\AttributeValue;
+
+   $smallSize = new AttributeValue();
+   $smallSize->setValue('Small');
+
+   $archetype = new Archetype();
+   $archetype->addAttribute($smallSize);
+
+   $product->setArchetype($archetype);
+
+.. note::
+   Doesn't matter if you use **Archetype** from this component,
+   the :doc:`/components/Archetype/index` or your custom. |br| Every model
+   implementing the :ref:`component_archetype_model_archetype-interface` is supported.
+
+Product attributes management
+-----------------------------
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Product\Model\Attribute;
+   use Doctrine\Common\Collections\ArrayCollection;
+
+   $attribute = new Attribute();
+
+   $colorGreen = new AttributeValue();
+   $colorRed = new AttributeValue();
+
+   $attributes = new ArrayCollection();
+
+   $attribute->setName('Color');
+
+   $colorGreen->setValue('Green');
+   $colorRed->setValue('Red');
+
+   $product->addAttribute($colorGreen);
+   $product->hasAttribute($colorGreen); // returns true
+   $product->removeAttribute($colorGreen);
+
+   $attributes->add($colorGreen);
+   $attributes->add($colorRed);
+   $product->setAttributes($attributes);
+
+   $product->hasAttributeByName('Color');
+   $product->getAttributeByName('Color'); // returns $colorGreen
+
+   $product->getAttributes();
+
+The ``getAttributes`` method returns an array, containing not
+only the color attributes created prior, but also every attribute value
+of the archetype assigned to the product so in this case the ``$smallSize``.
```

Comment:
Well, in Sylius - yes. With ProductBundle configured - true. But this does not happen automatically if you operate on raw components. Let's change it. It will return only color attributes. :)